src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -67,113 +69,143 @@ public static ParseUDFsResult ParseUDFsScript(string script, CultureInfo loc = n
+            ParseTrivia();
+            var ident = TokEat(TokKind.Ident);
+            if (ident == null)
+            {
+                return false;
+            }
 
-                TokEat(TokKind.ParenClose);
+            var args = new Dictionary<string, UDFArg>();
+            if (TokEat(TokKind.ParenOpen) == null)
+            {
+                return false;
+            }
 
+            while (_curs.TokCur.Kind != TokKind.ParenClose)
+            {
+                ParseTrivia();
+                var varIdent = TokEat(TokKind.Ident);
                 ParseTrivia();
-
                 if (TokEat(TokKind.Colon) == null)
                 {
-                    CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingColon);
                     break;
                 }
 
                 ParseTrivia();
-
-                var returnType = TokEat(TokKind.Ident);
-                if (returnType == null)
+                var varType = TokEat(TokKind.Ident);
+                ParseTrivia();
+                if (args.ContainsKey(varIdent.ToString()))
                 {
-                    CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingIdentifier);
-                    break;
+                    return false;
                 }
 
-                ParseTrivia();
-
-                if (TokEat(TokKind.Equ) == null)
+                args.Add(varIdent.ToString(), new UDFArg(varIdent.As<IdentToken>(), varType.As<IdentToken>()));
+                if (_curs.TokCur.Kind != TokKind.ParenClose && _curs.TokCur.Kind != TokKind.Comma)
                 {
-                    CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingEqual);
-                    break;
+                    CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingComma);
+                    return false;
                 }
+                else if (_curs.TokCur.Kind == TokKind.Comma)
+                {
+                    TokEat(TokKind.Comma);
+                }
+            }
 
-                ParseTrivia();
+            TokEat(TokKind.ParenClose);
 
-                // Parse body
-                while (_curs.TokCur.Kind != TokKind.Semicolon)
-                {
-                    // Check if we're at EOF before a semicolon is found
-                    if (_curs.TidCur == TokKind.Eof)
-                    {
-                        CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingSemicolon);
-                        return new ParseUDFsResult(udfs, _errors);
-                    }
+            ParseTrivia();
+
+            if (TokEat(TokKind.Colon) == null)
+            {
+                return false;
+            }
+
+            ParseTrivia();
 
-                    // Parse expression
-                    var result = ParseExpr(Precedence.None);
-                    udfs.Add(new UDF(ident.As<IdentToken>(), returnType.As<IdentToken>(), new HashSet<UDFArg>(args.Values), result));
+            var returnType = TokEat(TokKind.Ident);
```

Comment:
Might be worth allowing TokEat to skip the error, so we can create a more useful one in these cases.